Add examples for KVP search (#23)
ome/omero-guide-introduction: docs/search-omero.rst

#. Click on the Browse link |image2|\ in one search result line of the last image (in the right-hand part of the centre pane) to navigate back to the main webclient.

.. note::
    Spaces, stars, question marks and ``^`` characters are to be avoided in the Keys.

>     If you already have these characters in Keys in your OMERO, 

Reviewer: "OMERO server",

Author: fixed in https://github.com/ome/omero-guide-introduction/pull/23/commits/3e59c0df2de3a87681533e944bf1872f9a18b33c

    then try to replace them with underscores if you want to 
    use the Search functionality on them.

>     Spaces stars, question marks and ``^`` characters in Values 

Reviewer 2: ```suggestion
    Spaces, stars, question marks and ``^`` characters in Values 
```

>     are acceptable, but not preferable, see examples below.

Reviewer 2: ```suggestion
    are acceptable, but should be avoided if possible, see examples below.
```

Key-Value pairs search examples
-------------------------------

> Considering following setup of 12 separate images (image 1 through 12):

Reviewer 2: ```suggestion
Considering the following setup of 12 separate images (image 1 through 12):
```

> 1. Key ``GFP H2B`` with Value ``2 uM`` on image 1

Reviewer 3: This might be more readable in a table of: image id, name, key, value

Author: fixed in https://github.com/ome/omero-guide-introduction/pull/23/commits/86941d46d5c7573855222c80d53df5caae1446fe

2. Key ``GFP^H2B`` with Value ``2 uM`` on image 2
3. Key ``H2B`` with Value ``2`` on image 3
4. Key ``H2B`` with Value ``4`` on image 4
5. Key ``GFP-H2B`` with Value ``2-uM`` on image 5
6. Key ``GFP-H2B`` with Value ``2 uM`` on image 6
7. Key ``GFP_H2B`` with Value ``2_uM`` on image 7
8. Key ``GFP^H2B`` with Value ``2^uM`` on image 8
9. Image 9 named ``GFP`` with no Key-Value pairs
10. Image 10 named ``uM`` with no Key-Value pairs
11. Image 11 named ``H2B`` with no Key-Value pairs
12. Image 12 named ``2`` with no Key-Value pairs

Basic **Search** tab:

>    - ``GFP H2B:2 uM`` finds images 1,2,3,5,6,7,8,9,10. This is because the query is interpreted as ``GFP`` OR ``H2B:2`` OR ``uM``.

Reviewer 2: ```suggestion
   - ``GFP H2B:2 uM`` finds images 1,2,3,5,6,7,8,9,10. In that case, the query is interpreted as ``GFP`` OR ``H2B:2`` OR ``uM``.
```

>    - ``"GFP H2B":2 uM`` throws an error. Do not use quotes around Key names!

Reviewer 3: That this throws an error seems like quite the bug/limitation.

Author: Indeed. To explain, this error is "caught" and the user is confronted with a message

`` There was an error parsing your query. Colons ':' are reserved for searches of key-value annotations in the form: 'key:value'. ``

This behaviour is not only limiting, but also the message has a potential to exasperate the user. The user was using the colon in the right way, and this was their main purpose. The error message is highlighting the wrong culprit, and thus not hinting to the solution (the culprit here are the quotes, the colon should stay).
cc @will-moore

Author: @joshmoore created https://github.com/ome/omero-web/issues/373

>    - ``GFP H2B:"2 uM"`` finds images 1,2,5,6,7,8,9. This is because the query is interpreted as ``GFP`` OR ``H2B:2 uM`` which prevents finding of image 3 with Value ``2``.

Reviewer 2: ```suggestion
   - ``GFP H2B:"2 uM"`` finds images 1,2,5,6,7,8,9. In that case, the query is interpreted as ``GFP`` OR ``H2B:2 uM`` which prevents finding of image 3 with Value ``2``.
```

>    - ``GFP H2B:"2 uM"`` finds images 1,2,5,6,7,8,9. In that case, the query is interpreted as ``GFP`` OR ``H2B:2 uM`` which prevents finding of image 3 with Value ``2``.

Reviewer 3: Can any wildecards be added to key values?

Author: > Can any wildecards be added to key values?

Splitting the question. 
``Can any wildecards be added to keys?`` No. It throws the error https://github.com/ome/omero-guide-introduction/pull/23#discussion_r879399724
``Can any wildecards be added to Values?`` Yes, works intuitively. I will add an example.

Author: fixed in https://github.com/ome/omero-guide-introduction/pull/23/commits/e59ace6d928b18d326707a5851266b1fc759fd2d

>    - ``GFP^H2B:2 uM`` finds images 1,2,3,5,6,7,8,9,10. This is because the query is interpreted as ``GFP`` OR ``H2B:2`` OR ``uM``.

Reviewer 2: ```suggestion
   - ``GFP^H2B:2 uM`` finds images 1,2,3,5,6,7,8,9,10. In that case, the query is interpreted as ``GFP`` OR ``H2B:2`` OR ``uM``.
```

   - ``H2B:2`` finds image 3.
   - ``H2B:4`` finds image 4.
   - ``GFP-H2B:2 uM`` finds images 1,2,5,6,7,8,10.
   - ``GFP-H2B:2-uM`` finds images 5,6.
   - ``GFP-H2B:"2-uM"`` finds images 5,6.
   - ``GFP-H2B:"2 uM"`` finds images 5,6.
   - ``GFP-H2B:"2_uM"`` finds images 5,6.
   - ``GFP_H2B:2_uM`` finds image 7.
   - ``GFP^H2B:2^uM`` finds images 1,2,3,5,6,7,8,9,10.
   - ``GFP`` finds images 1,2,5,6,7,8,9.
   - ``GFP`` with checkbox ``Name`` under ``Restricted by Field`` section checked finds image 9.

**Advanced** tab:

>    - ``GFP^H2B:2^uM`` and ``GFP^H2B:2 uM`` throw an error in ``Advanced`` tab. This is because of the different interpretation of the ``^`` character between the basic ``Search`` and ``Advanced`` tabs.

Reviewer 2: ```suggestion
   - ``GFP^H2B:2^uM`` and ``GFP^H2B:2 uM`` throw an error in ``Advanced`` tab. This is due to the different interpretation of the ``^`` character between the basic ``Search`` and ``Advanced`` tabs.
```

>    - ``GFP^H2B:2^uM`` and ``GFP^H2B:2 uM`` throw an error in ``Advanced`` tab. This is due to the different interpretation of the ``^`` character between the basic ``Search`` and ``Advanced`` tabs.

Reviewer 3: Again, :+1: for capturing the current state, but this feels like it minimally needs a try/catch around it.

Author: That is the same error as in https://github.com/ome/omero-guide-introduction/pull/23#discussion_r879399724 - so indeed, it is tried and caught, but not much happiness is coming of it.

   - As there is no ``Name`` checkbox in the ``Advanced`` tab, use ``name:GFP`` instead, which finds image 9.

The behaviour for the rest of the query examples in ``Advanced`` tab is the same as listed above for the basic ``Search`` tab.

> 

Reviewer: From this, it sounds like the `Advanced` tab is the same as the regular search, except that it has a couple of limitations. I can see from the walk-through above that you can do `AND..` etc searches with the Advanced search, but I wonder if it's worth describing the difference between them or showing some more examples of what the Advanced search offers? Not sure if it's possible to illustrate that with the Images listed above?

Author: > ...what the Advanced search offers?

@will-moore What larger topics would you like to see covered ? The logical operators are covered, in my mind, this is the main strength of the Advanced tab. Do you maybe mean fuzzy search ? What else ?

Author: fixed in https://github.com/ome/omero-guide-introduction/pull/23/commits/9003cf5a5c92d00db3fc40b30d557f3a1cf0ffa5
